src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -214,6 +214,7 @@ public Connection getConnection() throws SQLException {
       properties.setProperty("password", dbPassword.value());
     }
     properties = addConnectionProperties(properties);
+    DriverManager.setLoginTimeout(60);
```

Comment:
what's the reasoning behind 60 seconds here?

Reply:
In the end, any number is a little arbitrary -- 60 just felt so conservative that no reasonable user would be waiting that long for a connection, and short enough that it does finally release our threads. I personally the 99.99% of connections will clear in under 10 seconds, but you never know with all the crazy routers, firewalls, proxies, etc. The main goal of the PR is to not block herder indefinitely, so I reasoned it is fine to be conservative. @wicknicks

Reply:
This happens during connector validation, right?

When a connector config is submitted to a distributed worker, it gets validated as soon as the worker has a chance to look at it. Then, if the worker isn't the leader, it forwards the config to the leader, which validates it _again_, and then finally writes it to the config topic. When all of this is done, the user's REST request is given a 200-level response.

Given that this means the connector config will possibly (even likely, since in any cluster of two or more workers you have a <=50% chance of hitting the leader with a request) be validated twice, and the REST request timeout in distributed mode is 90 seconds, maybe something like 40-45 seconds would be more appropriate here? That way we could avoid the scenario (which is possible with a driver connection timeout of 60 seconds) of the connection taking 60 seconds to succeed during the first worker's validation, then taking another 60 seconds to succeed during the leader's validation, then getting written to the config topic--even though at that point the user's REST request will have been met with a 500 timeout response.

Reply:
That's a reasonable suggestion -- 40s is a good idea. If @wicknicks  is onboard i'll set to 40 and we can merge

Reply:
yeah, @C0urante's suggestion makes sense. please add a comment to the doc, explaining this reasoning. thx in advance!